src/components/Download/DownloadOptionsForm/index.js

```diff
@@ -0,0 +1,114 @@
+import { useState } from 'react'
+import { useRouter } from 'next/router'
+import { Formik } from 'formik'
+import { Box, Form } from 'grommet'
+import { useDatasetManager } from 'hooks/useDatasetManager'
+import { useResponsive } from 'hooks/useResponsive'
+import { Button } from 'components/shared/Button'
+import { Row } from 'components/shared/Row'
+import { AdvanedOptions } from './AdvanedOptions'
+import { AdvancedOptionsToggle } from './AdvancedOptionsToggle'
+import { AggregateOptions } from './AggregateOptions'
+import { TransformationOptions } from './TransformationOptions'
+
+export const DownloadOptionsForm = ({
+  dataset = null,
+  buttonLabel = 'Download',
+  onSubmit = null
+}) => {
+  const { push } = useRouter()
+  const { dataset: datasetState, updateDataset } = useDatasetManager()
+  const { setResponsive } = useResponsive()
+  const [toggleAdvancedOption, setToggleAdvancedOption] = useState(false)
+  const selectedDataset = dataset || datasetState
+
+  const handleSubmitForm = async (options) => {
+    let pathname = '/download'
+
+    if (onSubmit) {
+      const response = await onSubmit() // TEMP
+      pathname = response
+    } else {
+      await updateDataset(selectedDataset.id, {
+        ...options,
+        data: selectedDataset.data
+      })
+    }
```

Comment:
Can you explain why you stubbed in the TEMP handling of `onSubmit`?

Reply:
Oh, this implementation was incomplete in this PR (in progress) but it was completed in the subsequent PR and the comment `// TEMP` was removed.

Please see the updated implementation [`src/components/Download/DownloadOptionsForm/index.js`](https://github.com/AlexsLemonade/refinebio-web/pull/199/files#diff-d5634ad7b9ba79f25a521e6c4c7d4c73a8fe76a34bce1f782b92686df7cf9562) (line 29) (https://github.com/AlexsLemonade/refinebio-web/pull/199)